app/uk/gov/hmrc/plasticpackagingtax/registration/models/genericregistration/PartnershipDetails.scala

```diff
@@ -16,34 +16,15 @@
 
 package uk.gov.hmrc.plasticpackagingtax.registration.models.genericregistration
 
-import play.api.libs.functional.syntax.{toFunctionalBuilderOps, unlift}
-import play.api.libs.json.{__, Format, Json, OFormat, Reads, Writes}
-import uk.gov.hmrc.plasticpackagingtax.registration.connectors.GrsEntityDetails
+import play.api.libs.json.{Json, OFormat}
+import uk.gov.hmrc.plasticpackagingtax.registration.forms.PartnershipTypeEnum.PartnershipTypeEnum
 
 case class PartnershipDetails(
-  sautr: String,
-  postcode: String,
-  override val registration: IncorporationRegistrationDetails
-) extends RegistrationDetails with GrsEntityDetails
+  partnershipType: PartnershipTypeEnum,
+  generalPartnershipDetails: Option[GeneralPartnershipDetails] = None
+  // TODO: other partnership type details here
+)
```

Comment:
I'm trying to follow the same pattern here as that already adopted for business types: ie. a type and a bunch of Optional type details fields.